url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/generator.java

```diff
@@ -0,0 +1,116 @@
+package utility;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.util.Random;
+import java.util.Scanner;
+
+public class generator {
+	public static void GeneratePointsofInteger(int range, int num, String filepath){
+		Random ramdom = new Random();
+		try {
+			PrintStream out = new PrintStream(new FileOutputStream(filepath));
+			for(int i = 0; i < num; i++){
+				int x = ramdom.nextInt(range), y = ramdom.nextInt(range);
+				out.println("1 " + (i + 1) + " " + x + " " + y + " " + x + " " + y);
+			}
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
+	}
+	
+	public static void GeneratePointsofInteger(int sx, int sy, int rangex, int rangey, int num, String filepath){
+		Random ramdom = new Random();
+		try {
+			PrintStream out = new PrintStream(new FileOutputStream(filepath));
+			for(int i = 0; i < num; i++){
+				int x = ramdom.nextInt(rangex), y = ramdom.nextInt(rangey);
+				out.println("1 " + (i + 1) + " " + (sx + x) + " " + (sy + y) + " " + (sx + x) + " " + (sy + y));
+			}
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
+	}
+	
+	public static void GeneratePointsofDouble(double sx, double sy, double mx, double my, int num, String filepath){
+		Random ramdom = new Random();
+		double rangex = mx - sx;
+		double rangey = my - sy;
+		try {
+			PrintStream out = new PrintStream(new FileOutputStream(filepath));
+			for(int i = 0; i < num; i++){
+				double x = ramdom.nextDouble() * rangex, y = ramdom.nextDouble() * rangey;
+				out.println((sx + x) + "\t" + (sy + y));
+			}
+			out.flush();
+			out.close();
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+	
+	public static void generatorofnear(int num, String filepath){
+		Scanner in;
+		Random ramdom = new Random();
+		try {
+			in = new Scanner(new FileInputStream(new File("source/ca.1000.dat")));
+			PrintStream out = new PrintStream(new FileOutputStream(filepath));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. Execution paths that do not contain closure statements were detected. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).